doc/locale_api.md

```diff
@@ -0,0 +1,97 @@
+# Legacy-free Locale Service
+
+> D-Installer currently have a separate Language service, although it's rather
+> simplistic. It just allows to set the language of the installed system using
+> Yast::Language.Set. And it's quite memory demanding for such an unimpressive
+> task.
+
+> That service would be a nice candidate to be rewritten from scratch with no
+> dependencies on YaST or Ruby. It's small enough and could give us a good
+> overview on how much can we save.
+
+Plan:
+1. take the systemd APIs as a sensible starting point.
+2. deviate only where we add value
+
+### Language and Keyboard
+
+#### Systemd
+
+The systemd API for Locale(Language) and Keyboard is this:
+(where the last boolean means Interactive, and the other boolean means Convert)
+
+```
+org.freedesktop.locale1 service
+/org/freedesktop/locale1 object
+NAME                    TYPE      SIG   RESULT/VALUE
+org.freedesktop.locale1 interface -     -
+.SetLocale              method    asb   -
+.SetVConsoleKeyboard    method    ssbb  -
+.SetX11Keyboard         method    ssssbb-
+(all properties are read-only and emit PropertiesChanged)
+.Locale                 property  as    1 "LANG=en_US.UTF-8"
+.VConsoleKeymap         property  s     "cz-lat2-us"
+.VConsoleKeymapToggle   property  s     ""
+.X11Layout              property  s     "cz,us"
+.X11Model               property  s     "pc105"
+.X11Options             property  s     "terminate:ctrl_alt_bksp,grp:shift_togg…
+.X11Variant             property  s     "qwerty,basic"
+```
+
+#### Design of Elementary Layer (lower)
+
+- Just use the systemd API, don't add any API of our own
+
+#### Design of Proposal Layer (uppper)
+
+- when setting the locale, adjust the proposed package selection and keyboard
+  accordingly. And timezone.
```

Comment:
important information here is what happens if user already changed keyboard. And another question is if we allow installation of multi localized environment. As current yast adds single locale, but I think in software selection you can add more. So if user of agama can say I want locale cs_CZ, but install also de_DE localization so user can select itself which to use. And if we decide to not support it, then we should document it with reason.

Reply:
**1.** you mean, first I change the keyboard from US to German, and then the language from English to Czech?

Should we override the user's choice of keyboard? Let's see the aspects of the choice:
1. does it waste a lot of user's work (for example, the whole partitioning layout should not be reproposed automatically)
2. is the change done secretly (even a trivial change should not be done if the GUI has no way of telling the user; for CLI it's fine as long as clearly documented)
3. is it reasonably easy in code and API?

I'm designing a better-detailed proposal layer with these considerations

Reply:
OK, addressed all of this in a recent commit